linkedList.js

```diff
@@ -3,6 +3,108 @@
  * @description  Main Module
  * @return {Object} an object exposing methods to be used to manipulate a linked list
  */
-function linkedListGenerator(){
+function linkedListGenerator() {
+  let head = null;
+  let tail = null;
+
+  function getHead() {
+    return head;
+  }
+
+  function getTail() {
+    if (tail === null) {
+      return null;
+    };
+
+    tail = head;
+    while (tail.next !== null) {
+      tail = tail.next;
+    }
+    return tail;
+  }
+
+  function add(value) {
+    let newNode = {};
+    newNode.value = value;
+    newNode.next = null;
+
+    if (head === null) {
+      newNode.next = head;
+      head = newNode;
+    } else {
+      let endOfNodeList = head;
+      while (endOfNodeList.next !== null) {
```

Comment:
Instead of doing a while loop here, you could use your getTail() function here instead.